dpnp/dpnp_iface_statistics.py

```diff
@@ -621,57 +640,157 @@ def mean(a, /, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
+       but it will probably be fully or partially sorted.
+       Default: ``False``.
+    keepdims : {None, bool}, optional
+        If ``True``, the reduced axes (dimensions) are included in the result
+        as singleton dimensions, so that the returned array remains
+        compatible with the input array according to Array Broadcasting
+        rules. Otherwise, if ``False``, the reduced axes are not included in
+        the returned array.
+        Default: ``False``.
+
+    Returns
+    -------
+    dpnp.median : dpnp.ndarray
+        A new array holding the result. If `a` has a floating-point data type,
+        the returned array will have the same data type as `a`. If `a` has a
+        boolean or integral data type, the returned array will have the
+        default floating point data type for the device where input array `a`
+        is allocated.
 
     See Also
     --------
     :obj:`dpnp.mean` : Compute the arithmetic mean along the specified axis.
     :obj:`dpnp.percentile` : Compute the q-th percentile of the data
                              along the specified axis.
 
+    Notes
+    -----
+    Given a vector ``V`` of length ``N``, the median of ``V`` is the
+    middle value of a sorted copy of ``V``, ``V_sorted`` - i.e.,
+    ``V_sorted[(N-1)/2]``, when ``N`` is odd, and the average of the
+    two middle values of ``V_sorted`` when ``N`` is even.
+
     Examples
     --------
     >>> import dpnp as np
     >>> a = np.array([[10, 7, 4], [3, 2, 1]])
+    >>> a
+    array([[10,  7,  4],
+           [ 3,  2,  1]])
     >>> np.median(a)
-    3.5
+    array(3.5)
+    >>> np.median(a, axis=0)
+    array([6.5, 4.5, 2.5])
+    >>> np.median(a, axis=1)
+    array([7.,  2.])
+    >>> np.median(a, axis=(0, 1))
+    array(3.5)
+    >>> m = np.median(a, axis=0)
+    >>> out = np.zeros_like(m)
+    >>> np.median(a, axis=0, out=m)
+    array([6.5,  4.5,  2.5])
+    >>> m
+    array([6.5,  4.5,  2.5])
+    >>> b = a.copy()
+    >>> np.median(b, axis=1, overwrite_input=True)
+    array([7.,  2.])
+    >>> assert not np.all(a==b)
+    >>> b = a.copy()
+    >>> np.median(b, axis=None, overwrite_input=True)
+    array(3.5)
+    >>> assert not np.all(a==b)
 
     """
 
-    x1_desc = dpnp.get_dpnp_descriptor(x1, copy_when_nondefault_queue=False)
-    if x1_desc:
-        if axis is not None:
-            pass
-        elif out is not None:
-            pass
-        elif overwrite_input:
-            pass
-        elif keepdims:
-            pass
+    dpnp.check_supported_arrays_type(a)
+    a_ndim = a.ndim
+    a_shape = a.shape
+    _axis = range(a_ndim) if axis is None else axis
+    _axis = normalize_axis_tuple(_axis, a_ndim)
+
+    if isinstance(axis, (tuple, list)):
+        if len(axis) == 1:
+            axis = axis[0]
         else:
-            result_obj = dpnp_median(x1_desc).get_pyobj()
-            result = dpnp.convert_single_elem_array_to_scalar(result_obj)
```

Comment:
We can remove `convert_single_elem_array_to_scalar` function, because it is not needed anymore.